src/lib/components/virtualized-table/sections/TableCells.svelte

```diff
@@ -1,31 +1,42 @@
 <script lang="ts">
-  import type { ProfileColumn } from "$lib/types";
+  import type { VirtualizedTableColumns } from "$lib/types";
   import Cell from "../core/Cell.svelte";
   import Row from "../core/Row.svelte";
+
   export let virtualColumnItems;
   export let virtualRowItems;
   export let rows;
-  export let columns: ProfileColumn[];
+  export let selectedIndex = [];
+  export let columns: VirtualizedTableColumns[];
   export let scrolling = false;
   export let activeIndex: number;
+
+  $: atLeastOneSelected = !!selectedIndex?.length;
 </script>
 
 {#each virtualColumnItems as column (column.key)}
   <Row>
     {#each virtualRowItems as row (`${row.key}-${column.key}`)}
+      {@const formattedValue =
+        rows[row.index]["__formatted_" + columns[column.index]?.name]}
       {@const value = rows[row.index][columns[column.index]?.name]}
       {@const type = columns[column.index]?.type}
       {@const rowActive = activeIndex === row?.index}
       {@const suppressTooltip = scrolling}
-
+      {@const rowSelected =
+        selectedIndex.findIndex((value) => row?.index === value) >= 0}
```

Comment:
I think for this case, the code is much more readable with the current structure. Moving this to script adds unnecessary reactive complexities.

Reply:
not going to block on this, but for reference i might try something like this:

```
  const getCellProps = (row, column) => {
    const value = rows[row.index][columns[column.index]?.name];

    return {
      value,
      formattedValue:
        rows[row.index]["__formatted_" + columns[column.index]?.name],
      type: columns[column.index]?.type,
      rowActive: activeIndex === row?.index,
      suppressTooltip: scrolling,
      barValue: columns[column.index]?.total
        ? value / columns[column.index]?.total
        : 0,
      rowSelected: selectedIndex.findIndex((tgt) => row?.index === tgt) >= 0,
    };
  };
</script>

{#each virtualColumnItems as column (column.key)}
  <Row>
    {#each virtualRowItems as row (`${row.key}-${column.key}`)}
      <Cell
        {row}
        {column}
        {atLeastOneSelected}
        {...getCellProps(row, column)}
        on:inspect
        on:select-item
      />
    {/each}
  </Row>
{/each}

```

Imo that's cleaner overall because you assemble the props object in  testable code (even though we are not testing today...), and can pass it as spread props to the component. Less code overall, testability improved, and less noise in the presentational component. Not a huge deal in this case, but i think this is a good pattern and practice to shoot for